### ISSUE TITLE
Error with bus::read function

### ISSUE THREAD

**Issue by wiired24** [adam-becker/psxact#12]
Sorry to bother you again, but I managed to track down and find the last critical error in the codebase.  Inside of the bus.hpp file you created a few functions one for reading and writing to the bus. The issue is that in dma_core.cpp the bus::read function gets called but when you declared the function in the bus.hpp header you specified 3 paramters.  I believe it was  (int width, uint32_t address, uint32_t data);
However when it's being called in dma_core.cpp it's only being passed two arguments rather than the 3 that were declared in the header file. So the Compiler is complaining that "bus::read does not take 2 arguments". This results in roughly 6 Errors all from the same issue. I'm unsure really how you wanna go about tackling this. I dunno if you had planned on inputting different width values for different function calls. Or just reusing a method that gets and returns the width.  Anyways Thank you in advance.

**Comment by wiired24** [comment]
Accoring to line 23 in bus.hpp it takes three though....

http://i.imgur.com/JRSZU4c.png see here

If only passing two is correct why then would the compiler be telling me bus::read function does not take 2 args.

**Comment by adam-becker** [comment]
Well, in _this_ code base, [read](https://github.com/adam-becker/psxact/blob/develop/src/bus.hpp#L18) is a function that takes 2 parameters. In the screen shot you've provided, I can see the green squiggle that means there isn't a corresponding implementation.

Your copy/paste job appears to be sloppy in this regard 😉

**Comment by wiired24** [comment]
It's fixed now. I just removed the third argument and the errors are resolved. btw I am somewhat new to git/github so I apologize that it's been taking a bit but  I have been trying to figure out how to add the repo as a submodule but the couple of tutorials I have looked at didn't work for me. Figured since you recommended it you could explain it to me ?

**Comment by wiired24** [comment]
For the fact that I fixed two critical issues in your codebase which resolved many build errors as well as added additional comments into psxbox not too bad for a copy and paste job eh ? ;)

**Comment by adam-becker** [comment]
Well, a submodule will only be necessary if you don't plan on going the library route we discussed earlier, wherein you consume PSXACT as a third-party dependency like you would with SDL.

I've thought over how I would expose PSXACT, and I think it would only require 2 function calls:

```c++
extern "C" {
  void psxact_init(const char *game_file_name, const char *bios_file_name);
  void psxact_run_for_one_frame(/* probably have to pass in controller data structs */);
}
```

A few more function calls may be required to allow you to retrieve the aural/visual data to render using whatever back-end you'd like. Assuming UWP, this is more than likely going to be DirectX11/XAudio2?

[I have opened an issue](https://github.com/adam-becker/psxact/issues/13) regarding this and discussion should continue there. Let me know what you think you'll need from me to make integrating PSXACT as easy as possible.
